### PR TITLE
build message models and its controllers

### PR DESCRIPTION
**Pull request by abdullah-alsharif**

### CODE REVIEW

File: server/controllers/message/createMessage.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const { Message } = require('../../models');
+
+const createMessage = async (req, res, next) => {
+  // const { sender, receiver, content } = req.params;
```
Comment (devstm) [inline server/controllers/message/createMessage.js]: remove this commint

File: server/controllers/message/createMessage.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  const { sender, receiver, content } = req.body;
+  try {
+    const message = await Message.create({
```
Comment (devstm) [inline server/controllers/message/createMessage.js:6]: { sender,
      receiver,
      content,
    } == req.body

File: server/controllers/message/deleteMessage.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const { Message } = require('../../models');
+
+const deleteMessage = async (req, res, next) => {
+  const { id } = req.params;
+  try {
+    const message = await Message.destroy({
```
Comment (devstm) [inline server/controllers/message/deleteMessage.js]: destroy function does not return the message object

File: server/controllers/message/deleteMessage.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const { Message } = require('../../models');
+
+const deleteMessage = async (req, res, next) => {
+  const { id } = req.params;
```
Comment (devstm) [inline server/controllers/message/deleteMessage.js:5]: check if message alredy exist

File: server/controllers/message/updateMessage.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+const { Message } = require('../../models');
+
+const updateMessage = async (req, res, next) => {
+  const { id } = req.params;
+  const { content } = req.body;
+  try {
+    const message = await Message.update(
```
Comment (devstm) [inline server/controllers/message/updateMessage.js:12]: update message does not return the message object
also check if message exist

File: server/middlewares/checkSenderReceiver.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+const CustomeError = require('../utils/helper/customeError');
+const { User } = require('../models');
+
+const checkSenderReceiver = async (req, res, next) => {
+  const { sender, receiver } = req.body;
+  try {
+    const senderExists = await User.findOne({
```
Comment (devstm) [inline server/middlewares/checkSenderReceiver.js:7]: extra comment 
use where [Op.OR]

**Review by devstm — CHANGES_REQUESTED** [review]

File: server/models/message.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
+const { DataTypes } = require('sequelize');
+const sequelize = require('../database/config/connection');
+
+const User = require('./user');
+
+const Message = sequelize.define('messages', {
+  sender: {
```
Comment (AyaQunoo) [inline server/models/message.js]: the relation will be already  done

**Review by AyaQunoo — APPROVED** [review]

File: server/middlewares/checkauth.js [hunk 8]
```diff
@@ -1,4 +1,4 @@
-const { CustomeError } = require('../utils/helper/customeError');
+const CustomeError = require('../utils/helper/customeError');
```
Comment (MohammedMahani) [inline server/middlewares/checkauth.js:1]: require custome error from the index in utils 
like : const { CustomeError } = require('../../utils'); in (deleteMessage.js)

File: server/routes/message.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+const messageRouter = require('express').Router();
+const {
+  createMessage,
+  readMessage,
+  deleteMessage,
+  updateMessage,
+} = require('../controllers');
+const checkAuth = require('../middlewares/checkauth');
```
Comment (MohammedMahani) [inline server/routes/message.js:8]: create index file to import the functions with short path

**Review by devstm — APPROVED** [review]

**Review by Basel-01 — APPROVED** [review]

**Pull request merged**